api/src/legacy_soap_api/applicants/services/get_opportunity_list_response.py

```diff
@@ -68,26 +116,38 @@ def _get_competitions(
     return db_session.execute(stmt).scalars().all()
 
 
-def _build_get_opportunity_list_response(competitions: list) -> GetOpportunityListResponse:
-    opportunity_details = (
-        list(map(_build_get_opportunity_details, competitions)) if competitions else []
+def _build_get_opportunity_list_response(
+    competitions: list, tcompetitions_map: dict
+) -> GetOpportunityListResponse:
+    if not competitions:
+        return GetOpportunityListResponse(opportunity_details=[])
+    return GetOpportunityListResponse(
+        opportunity_details=[
+            _build_get_opportunity_details(
+                competition, get_tinstructions_urls(competition, tcompetitions_map)
+            )
+            for competition in competitions
+        ]
```

Comment:
Suggestion - build the parts of the response conditionally, not the whole thing.
```suggestion
    opportunity_details = []
    if competitions:
        opportunity_details = [
            _build_get_opportunity_details(
                competition, get_tinstructions_urls(competition, tcompetitions_map)
            )
            for competition in competitions
        ]

   return GetOpportunityListResponse(opportunity_details=opportunity_details)
```